lib/exercism/authentication.rb

```diff
@@ -1,6 +1,36 @@
 class Authentication
   def self.perform(code, client_id, client_secret, target_profile)
-    id, username, email, avatar_url = Github.authenticate(code, client_id, client_secret)
-    User.from_github(id, username, email, avatar_url, target_profile)
+    new(code, client_id, client_secret, target_profile).authenticate
+  end
+
+  def initialize(code, client_id, client_secret, target_profile)
```

Comment:
This all seems to be happening in the middle of the authentication process, so that may be confusing me even too.